echopype/consolidate/api.py

```diff
@@ -174,3 +182,122 @@ def sel_interp(var):
+    RuntimeError
+        If ``ds`` does not have appropriate dimension lengths in comparison to ``echodata`` data
+    RuntimeError
+        If the provided ``waveform_mode``, ``encode_mode``, and ``pulse_compression`` are not valid
+    NotImplementedError
+        If an unknown ``beam_type`` is encountered during the split-beam calculation
+
+    Notes
+    -----
+    Split-beam angle data potentially exist for the the following echosounders depending on
+    the instrument configuration and recording setting:
+
+        - Simrad EK60 echosounder paired with split-beam transducers and
+          configured to store angle data
+        - Simrad EK80 echosounder paired with split-beam transducers and
+          configured to store angle data
+
+    For EK80 broadband data, the split-beam angles can be estimated from the complex data.
+    This functionality will be implemented in the future.
+    """
+
+    # ensure that echodata was produced by EK60 or EK80-like sensors
+    if echodata.sonar_model not in ["EK60", "ES70", "EK80", "ES80", "EA640"]:
+        raise RuntimeError(
+            "The sonar model that produced echodata does not have split-beam "
+            "transducers, split-beam angles cannot be added to ds!"
+        )
+
+    # check that the appropriate waveform and encode mode have been given
+    # and obtain the echodata group path corresponding to encode_mode
+    encode_mode_ed_group = check_waveform_encode_mode(
+        echodata, waveform_mode, encode_mode, pulse_compression
+    )
+
+    # check that ds at least has a channel dimension
+    if "channel" not in ds.variables:
+        raise RuntimeError("The input ds Dataset must have a channel dimension!")
+
+    # set ds_beam, select the same channels that are in ds (which is Sv)
+    ds_beam = echodata[encode_mode_ed_group].sel(channel=ds.channel.values)
+
+    # fail if ds and ds_beam do not have the same lengths for ping_time, range_sample, and channel
+    same_dim_lens = [
+        ds_beam.dims[dim] == ds.dims[dim] for dim in ["channel", "ping_time", "range_sample"]
+    ]
+    if not same_dim_lens:
+        raise RuntimeError(
+            "Input ds does not have the same dimension " "lengths as all dimensions in ds_beam!"
+        )
+
+    # obtain split-beam angles from
+    if (waveform_mode == "CW") and (encode_mode == "power") and (pulse_compression is False):
+        theta_fc, phi_fc = _get_splitbeam_angle_power_CW(ds_beam=ds_beam)
+    elif (waveform_mode == "CW") and (encode_mode == "complex") and (pulse_compression is False):
+        theta_fc, phi_fc = _get_splitbeam_angle_complex_CW(ds_beam=ds_beam)
+    elif (waveform_mode == "BB") and (encode_mode == "complex") and (pulse_compression is False):
+        theta_fc, phi_fc = _get_splitbeam_angle_complex_BB_nopc(ds_beam=ds_beam)
+    elif (waveform_mode == "BB") and (encode_mode == "complex") and (pulse_compression is True):
+        theta_fc, phi_fc = _get_splitbeam_angle_complex_BB_pc(ds_beam=ds_beam)
```

Comment:
I suggest the following for clarify because you already have a comprehensive check of the combination of input arguments in `echodata.simrad::_check_mode_input_without_data`
```suggestion
    # obtain split-beam angles from
    # CW mode data
    if waveform_mode == "CW":
       if encode_mode == "power":  # power data
          theta_fc, phi_fc = _get_splitbeam_angle_power_CW(ds_beam=ds_beam)
       else:  # complex data
          theta_fc, phi_fc = _get_splitbeam_angle_complex_CW(ds_beam=ds_beam)
    # BB mode data
    else:
       if pulse_compression:  # with pulse compression
          theta_fc, phi_fc = _get_splitbeam_angle_complex_BB_pc(ds_beam=ds_beam)
       else:  # without pulse compression
          theta_fc, phi_fc = _get_splitbeam_angle_complex_BB_nopc(ds_beam=ds_beam)          
```

Reply:
I agree, this reduction is nice and my check should catch that the appropriate combination has been provided.